templates/user_form2.tt.html

```diff
@@ -0,0 +1,241 @@
+<script>
+
+    function normalize_string_value(inputfield) {
+
+    var value = inputfield.value.toLowerCase();
+
+    value = value.replace(new RegExp(" ", 'g'),"-");
+    value = value.replace(new RegExp("[àáâãäå]", 'g'),"a");
+    value = value.replace(new RegExp("æ", 'g'),"ae");
+    value = value.replace(new RegExp("ç", 'g'),"c");
+    value = value.replace(new RegExp("[èéêë]", 'g'),"e");
+    value = value.replace(new RegExp("[ìíîï]", 'g'),"i");
+    value = value.replace(new RegExp("ñ", 'g'),"n");
+    value = value.replace(new RegExp("[òóôõö]", 'g'),"o");
+    value = value.replace(new RegExp("œ", 'g'),"oe");
+    value = value.replace(new RegExp("[ùúûü]", 'g'),"u");
+    value = value.replace(new RegExp("[ýÿ]", 'g'),"y");
+    value = value.replace(new RegExp("[^a-zA-Z0-9-]", 'g'),"-");
+    value = value.replace(new RegExp("-+", 'g'),"-");
+    value = value.replace(new RegExp("^-"),"");
+
+    inputfield.value = value;
+    }
+</script>
+
+[% IF action == 'display' || action == 'none' %]
+[% INCLUDE 'error_list.tt.html' %]
+[% END %]
+
+[% IF action == 'display' %]
+
+    <!-- Start form -->
+
+    <form method="post" action="/cgi/user2.pl" enctype="multipart/form-data">
+		[% FOREACH section IN sections %]
+			[% IF user OR section.id == "user" %]
```

Comment:
[% IF user OR section.id == "user" %]

--> there's no "user" variable in the template data $template_ref

in the org template, there is a similar line, but only for administrators:

[% IF admin OR section.id != "admin" %]

--> this is just to hide the "admin" section for administrators.  The "admin" variable is set in the template_data.

Reply:
`user` is the section id for the first section. every section has a different unique ID. These IDs are helping to distinguish the sections from each other.